include/aspect/prescribed_stokes_solution/interface.h

```diff
@@ -0,0 +1,175 @@
+/*
+  Copyright (C) 2011 - 2015 by the authors of the ASPECT code.
+
+  This file is part of ASPECT.
+
+  ASPECT is free software; you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation; either version 2, or (at your option)
+  any later version.
+
+  ASPECT is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with ASPECT; see the file doc/COPYING.  If not see
+  <http://www.gnu.org/licenses/>.
+*/
+
+
+
+#ifndef __aspect__prescribed_stokes_solution_interface_h
+#define __aspect__prescribed_stokes_solution_interface_h
+
+#include <aspect/plugins.h>
+
+#include <deal.II/base/point.h>
+#include <deal.II/base/parameter_handler.h>
+
+#include <deal.II/base/function.h>
+
+namespace aspect
+{
+  /**
+   * A namespace in which we define everything that has to do with defining
+   * the prescribed Stokes solution.
+   *
+   * @ingroup PrescribedStokesSolution
+   */
+  namespace PrescribedStokesSolution
+  {
+    using namespace dealii;
+
+    /**
+     * This plug in allows the user to prescribe a Stokes solution and can be thought of as
+     * velocity and pressure's equivalent of the initial conditions plug-in.
+     *
+     * Note: Only used if solver type  is "Advection only".
+     *
+     * @ingroup PrescribedStokesSolution
+     */
+    template <int dim>
+    class Interface
+    {
+      public:
+        /**
+         * Destructor. Made virtual to enforce that derived classes also have
+         * virtual destructors.
+         */
+        virtual ~Interface();
+
+        /**
+         * Initialization function. This function is called once at the
+         * beginning of the program after parse_parameters is run and after the
+         * SimulatorAccess (if applicable) is initialized.
+         */
+        virtual
+        void
+        initialize ();
+
+        /**
+         * Given a position @p p, fill in desired velocity and pressure at that point into @p value, which will have dim+1 components.
+         */
+        virtual
+        void   solution (const Point<dim>   &p, Vector<double>     &value) const = 0;
```

Comment:
I am not too happy about this function naming scheme. In all other parts of aspect, the functions are named after their property (temperature, specific_heating_rate, ...), so this should be velocity_pressure, or at least stokes_solution. Also when I read solution I always think of our solution vector. 
Independent of that, would it be possible to return a Vector<double> instead of changing a preexisting Vector? This would reduce the number of parameters. Or does this not work, because of the way it is called later on?